### PR TITLE
Lambda UX: Add remaining fields and logic to config editor form

### PR DESCRIPTION
**Pull request by gsweetwood**
## Description

Most properties of an AwsSamDebuggerConfiguration are now part of the form. Various fixes are also included

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the changelog using the script `npm run newChange`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by bryceitoc9** [comment]
Looks like there's an issue with the HTTP methods in the tests (returning undefined instead of one of the defined strings). I think that's the only hard flag that we have that's causing the tests to break, mind taking a look?

File: src/lambda/vue/DONOTUSE.vue [hunk 1]
```diff
@@ -7,18 +7,19 @@
```
Comment (bryceitoc9) [inline src/lambda/vue/DONOTUSE.vue:7]: Can we .gitignore this file and delete it from the repo?

File: src/lambda/vue/samInvokeVue.ts [hunk 2]
```diff
@@ -14,36 +15,81 @@ import {
...
+            localArguments: undefined,
+            skipNewImageCheck: false,
+            ...(existingConfig?.sam ? existingConfig.sam : {}),
+            template: {
+                parameters: (existingConfig?.sam?.template?.parameters ? existingConfig.sam.template.parameters : {})
+            }
+        },
+        api: {
+            path: '',
+            httpMethod: "",
```
Comment (bryceitoc9) [inline src/lambda/vue/samInvokeVue.ts:83]: Nit: single quotes

File: src/lambda/vue/samInvokeVue.ts [hunk 3]
```diff
@@ -109,61 +184,92 @@ export const Component = Vue.extend({
...
+            this.resetJsonErrors()
+            vscode.postMessage({
+                command: 'getTemplate',
+            })
+        },
+        toggleShowAllFields() {
+            this.showAllFields = !this.showAllFields
+        },
+        stringToBoolean(field: string){
+            if (field === "True") {
```
Comment (bryceitoc9) [inline src/lambda/vue/samInvokeVue.ts:230]: Nit: this entire function can be shortened to `return field === 'True'`

Unsure what happens if the `else if` isn't triggered.

File: src/lambda/vue/samInvokeVue.ts [hunk 4]
```diff
@@ -287,39 +399,129 @@ export const Component = Vue.extend({
...
+                    <label for="timeoutSec">Timeout (s)</label>
+                    <input type="number" v-model.number="launchConfig.lambda.timeoutSec" >
+                </div>
+                <!-- <div class="config-item">
+                    <label for="pathMappings">Path Mappings</label>
+                    <input type="text" v-model="launchConfig.lambda.pathMappings" >
+                </div> -->
+                <h3>sam</h3>
+                <div class="config-item">
+                    <label for="buildArguments">Build Arguments</label>
```
Comment (bryceitoc9) [inline src/lambda/vue/samInvokeVue.ts:457]: Maybe add a note for the array fields that a comma separated list of args should be provided.

File: src/lambda/vue/samInvokeVue.ts [hunk 5]
```diff
@@ -109,61 +184,92 @@ export const Component = Vue.extend({
             },
             deep: true,
         },
-        payload: function (newval: string) {
-            this.resetJsonError()
+        payload: function (newval: {value: string, errorMsg: string}) {
```
Comment (bryceitoc9) [inline src/lambda/vue/samInvokeVue.ts:187]: Just so I'm clear, how is this errorMsg set?

**Review by bryceitoc9 — APPROVED** [review]
Mainly nits. If this is working in your tests, we can probably fix them later (thanks, beta label!)

File: src/lambda/vue/DONOTUSE.vue (hunk 1 above)
Comment (gsweetwood) [reply src/lambda/vue/DONOTUSE.vue:7]: deleted

File: src/lambda/vue/samInvokeVue.ts (hunk 2 above)
Comment (gsweetwood) [reply src/lambda/vue/samInvokeVue.ts:83]: Changed

File: src/lambda/vue/samInvokeVue.ts (hunk 4 above)
Comment (gsweetwood) [reply src/lambda/vue/samInvokeVue.ts:457]: Added a placeholder: Enter as a comma separated list

File: src/lambda/vue/samInvokeVue.ts (hunk 3 above)
Comment (gsweetwood) [reply src/lambda/vue/samInvokeVue.ts:230]: refactored

File: src/lambda/vue/samInvokeVue.ts (hunk 5 above)
Comment (gsweetwood) [reply src/lambda/vue/samInvokeVue.ts:187]: errorMsg is set when attempting to parse the string into JSON.  This happens during save or invoke.

**Review by bryceitoc9 — APPROVED** [review]

**Pull request merged**